### PR TITLE
Rename dottedCircleFilter module

### PR DESCRIPTION
**Pull request by madig**
https://github.com/googlefonts/ufo2ft/pull/715 was wrong actually and broke main. I decided to rename the filter module instead and put in a redirection stub.

Closes https://github.com/googlefonts/ufo2ft/issues/707.

### CODE REVIEW

**Comment by khaledhosny** [comment]
I suggest renaming then adding the old name in two different commits so that git can still track file history.

**Comment by anthrotype** [comment]
please do what khaled said, otherwise LGTM (merge when done thanks)

**Comment by madig** [comment]
Ok, will do.

Hm, user warnings aren't displayed to users by default. I'll subclass and log a regular warning.

**Comment by anthrotype** [comment]
you mean, `DeprecationWarnings` aren't displayed to users by default.. the default `UserWarnings` should be

**Comment by madig** [comment]
Uh, well, do you want me to use UserWarnings instead?

**Comment by anthrotype** [comment]
> you want me to use UserWarnings instead?

instead of DeprecationWarning, yes. What else do you suggest instead?

**Comment by madig** [comment]
Eh, either way is fine, a logging.warning just looks neater. Updated to use a UserWarning.

File: Lib/ufo2ft/filters/dottedCircleFilter.py [hunk 1]
```diff
@@ -1,262 +1,21 @@
...
-sidebearings
-    When drawing a dotted circle, additional horizontal space in units around
-    the dotted circle.
-dots
-    Number of dots in the circle.
+"""NOTE: This is a redirection stub, because the original module was misnamed.
 
+It ended in "Filter", which messed with the assumptions in ``getFilterClass``.
+Make an alias for the class here to steer the importer in the right
+direction.
```
Comment (anthrotype) [inline Lib/ufo2ft/filters/dottedCircleFilter.py:5]: I don't really like this `DottedCircleFilterFilter` hack.. I think you need to issue a warnings.warn if the deprecated `dottedCircleFilter` module gets imported, so move this warning to the module scope. You don't need a `*FilterFilter` subclass...

Comment (anthrotype) [reply Lib/ufo2ft/filters/dottedCircleFilter.py:5]: Just ignore `Filter` suffix in the filter name if present, I think you did that at some point but then reverted (https://github.com/googlefonts/ufo2ft/pull/715)

**Comment by anthrotype** [comment]
> either way is fine, a logging.warning just looks neater.

it depends who the message is directed to. If it's the non-developer user of a tool that uses ufo2ft, then logging should be used; if it's a developer that imports ufo2ft from their own code, you should use warnings module

**Comment by madig** [comment]
Ok. Warn on import is harder to test because we import once but test twice with FontClass, so once it suceeds but then no more :/

**Review by anthrotype — APPROVED** [review]

**Pull request merged**